src/components/Form/Form.tsx

```diff
@@ -53,6 +59,14 @@ export const Form: FC<FormProps> = ({ activity }) => {
       },
     })
     await onSubmit(response)
+    removeLocalItem()
+  }
+
+  const handleOnAnswersChange = (response: Record<string, any>): void => {
+    if (JSON.stringify(response) === storedLocalValue) {
+      return
+    }
+    setLocalValue(JSON.stringify(response))
```

Comment:
Minor suggestion: stringify only once for simplicity

```suggestion
  const handleOnAnswersChange = (response: Record<string, any>): void => {
    const stringifiedResponse = JSON.stringify(response)
    if (stringifiedResponse === storedLocalValue) {
      return
    }
    setLocalValue(stringifiedResponse)
```